### PR TITLE
fix: blocklisting peers after waking up from hibernation

### PR DESCRIPTION
**Pull request by mrekucci**
Stops the internal clock of the blocker instance during hibernation so that the peer list is not blocked upon wakeup.

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2857)

### CODE REVIEW

File: pkg/blocker/blocker.go [hunk 1]
```diff
@@ -33,11 +39,17 @@ type Blocker struct {
 }
 
 func New(dis p2p.Blocklister, flagTimeout, blockDuration, wakeUpTime time.Duration, callback func(swarm.Address), logger logging.Logger) *Blocker {
+	if flagTimeout <= resolution {
+		panic(fmt.Errorf("flag timeout %v cannot be equal to or lower then the clock resolution %v", flagTimeout, resolution))
+	}
+	if wakeUpTime < resolution {
+		panic(fmt.Errorf("wakeup time %v cannot be lower then the clock resolution %v", wakeUpTime, resolution))
```
Comment (notanatol) [inline pkg/blocker/blocker.go:46]: wouldn't panicking here potentially cause inconsistencies in the store?

Comment (mrekucci) [reply pkg/blocker/blocker.go]: These panics basically mean programmer error. It makes no sense to check for flagged peers more often than the sequencer resolution rate.

Comment (notanatol) [reply pkg/blocker/blocker.go]: right, but would panicking trigger the shutdown sequence? or would it just quick the process?

Comment (mrekucci) [reply pkg/blocker/blocker.go]: Will crash the process.

**Review by notanatol — APPROVED** [review]

**Review by janos — APPROVED** [review]
👍

**Review by aloknerurkar — APPROVED** [review]

**Pull request merged**